aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -323,6 +323,9 @@ Tensor _histc_cuda_template(
     maxvalue = maxvalue + 1;
   }
 
+  TORCH_CHECK(!(THCNumerics<input_t>::isinf(minvalue) || THCNumerics<input_t>::isinf(maxvalue) || THCNumerics<input_t>::isnan(minvalue) || THCNumerics<input_t>::isnan(maxvalue)), "range of [", minvalue, ", ", maxvalue, "] is not finite");
+  TORCH_CHECK(minvalue < maxvalue, "max must be larger than min");
+
```

Comment:
rocm build is failing with:
```
18:29:27 [ 87%] Linking CXX executable ../bin/generate_proposals_op_gpu_test
18:29:29 ../lib/libtorch.so: undefined reference to `isinf(float)'
18:29:29 ../lib/libtorch.so: undefined reference to `isnan(float)'
18:29:29 ../lib/libtorch.so: undefined reference to `isnan(double)'
18:29:29 ../lib/libtorch.so: undefined reference to `isinf(double)'
18:29:29 clang: [0;1;31merror: [0mlinker command failed with exit code 1 (use -v to see invocation)[0m
```
To fix this, I think we need to `#include <THC/THCNumerics.cuh>` at the top of this file after `#include <ATen/cuda/CUDAApplyUtils.cuh>`, taking inspiration from the `aten/src/ATen/native/cuda/MultinomialKernel.cu` file.

Reply:
oh okay 👌 thanks!!